serviceaccounts.rb

```diff
@@ -34,35 +38,31 @@ def existing_file_account(path)
 
   def run()
     common = Common.new
-    ENV["GOOGLE_APPLICATION_CREDENTIALS"] = @path
-    service_account = @service_account
-    if not service_account
-      service_account = "#{@project}@appspot.gserviceaccount.com"
-    end
-    if service_account == existing_file_account(@path)
+    ENV["GOOGLE_APPLICATION_CREDENTIALS"] = @keyfile_path
+    if @service_account == existing_file_account(@keyfile_path)
       # Don't generate another key if this account is already active. This can
       # happen for nested service account contexts, for example.
-      common.status "Attaching to existing keyfile @ #{@path}"
-      yield
+      common.status "Attaching to existing keyfile @ #{@keyfile_path}"
+      yield(self)
       return
     end
 
-    if service_account == "all-of-us-workbench-test@appspot.gserviceaccount.com"
-      unless File.exists?(@path)
```

Comment:
FYI I removed this check - this was a bug and probably one of the causes of https://precisionmedicineinitiative.atlassian.net/browse/RW-1473